### ISSUE TITLE
IDE0180 false positive (resulting in non-compilable code)

### ISSUE THREAD

**Issue by kindermannhubert** [dotnet/roslyn#66427]
**Version Used**: 
VS 17.4.4

**Steps to Reproduce**:
```C#
Span<int> v0 = stackalloc int[n];
Span<int> v1 = stackalloc int[n];

var vTmp = v0;
v0 = v1;
v1 = vTmp;
```

or more generally:

```C#
ref struct S { }

S v0 = default;
S v1 = default;

var vTmp = v0;
v0 = v1;
v1 = vTmp;
```

**Diagnostic Id**:
IDE0180 - use tuple to swap values

**Expected Behavior**:
IDE0180 is not reported because tuples cannot be used with ref-structs.

**Actual Behavior**:
IDE0180 is reported and the resulting code after the suggested fix does not compile:
![image](https://user-images.githubusercontent.com/11704036/212475110-34931811-a8d0-4e57-9dc0-7c2969b99c49.png)

**Comment by Youssef1313** [comment]
cc @kinkeadian25

**Comment by Youssef1313** [comment]
The analyzer is currently written using `RegisterSyntaxNodeAction` and doesn't do any semantic model calls.

I think now it will require semantic model calls. @mavasani is it better for performance to rewrite this analyzer using IOperation?
